doc/src/modules/combinatorics/fp_groups.rst

```diff
@@ -0,0 +1,247 @@
+Can be called with different public versions of ``FreeGroup`` i.e
+``free_group``, ``vfree_group`` and ``xfree_group``.
+
+Methods for manipulating Words
+------------------------------
+
+This section describes some basic access functions for *words*. These operations apply
+to words of both ``FreeGroup`` as well as that of arbitrary ``FpGroup``.
+
+``copy``
+    return a
+
+``is_identity``
+
+``array_form``
+    Returns one of the internal use representation of ``word``.
+
+``letter_form``
+
+index
+
+ext_rep
+
+contains
+
+eliminate_word
+
+``in``
+
+``len(w)``
+    The lenth of word ``w``.
+
+
+Comparison of words
+-------------------
+
+Coset Enumeration: The Todd-Coxeter Algorithm
+=============================================
+
+This section describes the use of coset enumeration techniques in SymPy. The
+algorithm used for coset enumeration procedure is Todd-Coxeter algorithm and
+is developed in Sympy using [Ho05] and [CDHW73]. The reader should consult
+[CDHW73] and [Hav91] for a general description of the algorithm.
+
+We have two strategies of coset enumeration *relator-based* and
+*coset-table based* and the two have been implemented as
+``coset_enumeration_r``, ``coset_enumeration_c`` respectively. The two
+strategies differ in the way they make new definitions for the cosets.
+
+Though from the user point of view it is suggested to rather use the
+``.coset_enumeration`` method of ``FpGroup`` and specify the ``strategy``
+argument
+
+``strategy``:
+    (default="relator_based") specifies the strategy of coset
+    enumeration to be used, possible values are *"relator_based"* or
+    *"coset_table_based"*.
+
+CosetTable
+----------
+
+Class used to manipulate the information regarding the coset enumeration of
+the finitely presented group ``G`` on the cosets of the subgroup ``H``.
+
+Basically a *coset table* ``CosetTable(G,H)``, is the permutation representation
+of the finitely presented group on the cosets of a subgroup. Most of the set
+theoretic and group functions use the regular representation of ``G``, i.e.,
+the coset table of ``G`` over the trivial subgroup.
+
+The actual mathematical coset table is obtained using ``.table`` attribute and
+is a list of lists. For each generator ``g`` of ``G`` it contains a coloumn and
+the next coloumn corresponds to ``g**-1`` and so on for other generators, so in
+total it has ``2*G.rank()`` coloumns. Each coloumn is simply a list of integers.
+If ``l`` is the generator list for the generator `g` and if ``l[i] = j`` then
+generator ``g`` takes the coset `i` to the coset `j` by multiplication from the
+right.
+
+For finitely presented groups, a coset table is computed by a Todd-Coxeter
+coset enumeration. Note that you may influence the performance of that
+enumeration by changing the values of the variable
+``CosetTable.coset_table_max_limit``.
+
+Attributes of CosetTable
+------------------------
+
+For ``CosetTable(G, H)`` where ``G`` is the group and ``H`` is the subgroup.
+
+* ``p``:
```

Comment:
This could be omitted. It is strictly for internal use. `omega` contains the important information.